packages/block-library/src/cover-image/index.js

```diff
@@ -117,146 +127,171 @@ export const settings = {
-					<InspectorControls>
-						<PanelBody title={ __( 'Cover Image Settings' ) }>
-							<ToggleControl
-								label={ __( 'Fixed Background' ) }
-								checked={ !! hasParallax }
-								onChange={ toggleParallax }
-							/>
-							<RangeControl
-								label={ __( 'Background Opacity' ) }
-								value={ dimRatio }
-								onChange={ setDimRatio }
-								min={ 0 }
-								max={ 100 }
-								step={ 10 }
-							/>
-						</PanelBody>
-					</InspectorControls>
-				) }
-			</Fragment>
-		);
-
-		if ( ! url ) {
-			const hasTitle = ! RichText.isEmpty( title );
-			const icon = hasTitle ? undefined : 'format-image';
-			const label = hasTitle ? (
-				<RichText
-					tagName="h2"
-					value={ title }
-					onChange={ ( value ) => setAttributes( { title: value } ) }
-					inlineToolbar
-				/>
-			) : __( 'Cover Image' );
+					</Fragment>
+				);
+			}
 
 			return (
 				<Fragment>
 					{ controls }
-					<MediaPlaceholder
-						icon={ icon }
-						className={ className }
-						labels={ {
-							title: label,
-							name: __( 'an image' ),
-						} }
-						onSelect={ onSelectImage }
-						accept="image/*"
-						allowedTypes={ ALLOWED_MEDIA_TYPES }
-						notices={ noticeUI }
-						onError={ noticeOperations.createErrorNotice }
-					/>
+					<div
+						data-url={ url }
+						style={ style }
+						className={ classes }
+					>
+						{ ( ! RichText.isEmpty( title ) || isSelected ) && (
+							<RichText
+								tagName="p"
+								className="wp-block-cover-image-text"
+								placeholder={ __( 'Write title…' ) }
+								value={ title }
+								onChange={ ( value ) => setAttributes( { title: value } ) }
+								inlineToolbar
+							/>
+						) }
+					</div>
 				</Fragment>
 			);
 		}
-
-		return (
-			<Fragment>
-				{ controls }
-				<div
-					data-url={ url }
-					style={ style }
-					className={ classes }
-				>
-					{ ( ! RichText.isEmpty( title ) || isSelected ) && (
-						<RichText
-							tagName="p"
-							className="wp-block-cover-image-text"
-							placeholder={ __( 'Write title…' ) }
-							value={ title }
-							onChange={ ( value ) => setAttributes( { title: value } ) }
-							inlineToolbar
-						/>
-					) }
-				</div>
-			</Fragment>
-		);
-	} ),
+	),
 
 	save( { attributes, className } ) {
-		const { url, title, hasParallax, dimRatio, align, contentAlign } = attributes;
+		const { url, title, hasParallax, dimRatio, align, contentAlign, overlayColor, customOverlayColor } = attributes;
+		const overlayColorClass = getColorClassName( 'background-color', overlayColor );
 		const style = backgroundImageStyles( url );
+		if ( ! overlayColorClass ) {
+			style.backgroundColor = customOverlayColor;
+		}
+
 		const classes = classnames(
 			className,
 			dimRatioToClass( dimRatio ),
 			{
 				'has-background-dim': dimRatio !== 0,
 				'has-parallax': hasParallax,
 				[ `has-${ contentAlign }-content` ]: contentAlign !== 'center',
+				[ overlayColorClass ]: !! overlayColor,
```

Comment:
This can just be an argument to `classnames`, omitted if falsy, if `overlayColorClass` can be the boolean measure of if a custom color is assigned. Above we consider it to be (`if ( ! overlayColorClass ) {`), here we use `overlayColor`.